nixos/lib/test-driver/src/test_driver/machine/__init__.py

```diff
@@ -93,20 +93,25 @@ def make_command(args: list) -> str:
     return " ".join(map(shlex.quote, (map(str, args))))
 
 
-def retry(fn: Callable, timeout: int = 900) -> None:
-    """Call the given function repeatedly, with 1 second intervals,
-    until it returns True or a timeout is reached.
+def retry(fn: Callable, timeout: int = 900, interval: float = 1.0) -> None:
```

Comment:
Two little remarks that might help readability:

- `timeout` could be `timeout_seconds` so to know what unit we use, noone needs to read the implementation or search for it specifically in the docs. I know we did not use a unit in the name before, but it still seems like a reasonable upgrade.
- `interval` could be removed and we simply stick to a second. or we could also add the unit. e.g. `interval_seconds`

Reply:
Opted to remove the interval here and focus on the monotonic aspects (as there's already a `seconds_interval` in `driver.py`).